src/main/java/com/google/api/codegen/transformer/SampleTransformer.java

```diff
@@ -211,34 +212,45 @@ public void generateSamples(
                     .build());
       }
 
-      for (ValueSetAndTags valueSet : matchingValueSets) {
-        InitCodeView initCodeView =
-            sampleGenerator.generate(
-                initContext != null
-                    ? initContext
-                    : createInitCodeContext(
-                        context,
-                        fieldConfigs,
-                        initCodeOutputType,
-                        valueSet.values().getParametersList(),
-                        valueSet.values().getSampleArgsList()));
-        List<OutputSpec> outputs = valueSet.values().getOnSuccessList();
+      for (ValueSetAndTags setAndTag : matchingValueSets) {
+        // Don't overwrite the initContext in outer scope.
+        InitCodeContext thisContext = initContext;
+        SampleValueSet valueSet = setAndTag.values();
+        if (thisContext == null) {
+          // TODO(pongad): Check that the attributes don't overlap.
```

Comment:
I would explain what you mean by overlap. What we discussed was not having any part of the request object described by more than one `SampleInitAttribute`